onnxscript/function_libs/torch_aten/ops/nn.py

```diff
@@ -769,16 +770,81 @@ def aten_multilabel_margin_loss_forward(
     raise NotImplementedError()
 
 
+@torch_op("aten::nll_loss", trace_only=True)
 def aten_nll_loss(
-    self: TensorType,
-    target: TensorType,
-    weight: Optional[TensorType] = None,
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: Optional[TFloat] = None,
     reduction: int = 1,
-    ignore_index: INT64 = -100,
-) -> TensorType:
+    ignore_index: int = -100,
+) -> TFloat:
     """nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor"""
 
-    raise NotImplementedError()
+    reduction_vals = ["none", "mean", "sum"]
+    reduction_str = reduction_vals[reduction]
+
+    if op.OptionalHasElement(weight):
+        result = _aten_nll_loss_weight(
+            self, target, weight, reduction=reduction_str, ignore_index=ignore_index
+        )
+    else:
+        result = _aten_nll_loss_none(
+            self, target, reduction=reduction_str, ignore_index=ignore_index
+        )
```

Comment:
Is it possible to merge these two functions?

Reply:
combined.